dashboard/src/main/home/infrastructure/components/DeployList.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Context } from "shared/Context";
+import api from "shared/api";
+import styled from "styled-components";
+import Loading from "components/Loading";
+import { Operation, OperationStatus, OperationType } from "shared/types";
+import { readableDate } from "shared/string_utils";
+import Placeholder from "components/Placeholder";
+
+type Props = {
+  infra_id: number;
+};
+
+const DeployList: React.FunctionComponent<Props> = ({ infra_id }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const [operationList, setOperationList] = useState<Operation[]>([]);
+  const { currentProject, setCurrentError } = useContext(Context);
+
+  useEffect(() => {
+    api
+      .listOperations(
+        "<token>",
+        {},
+        {
+          project_id: currentProject.id,
+          infra_id: infra_id,
+        }
+      )
+      .then(({ data }) => {
+        if (!Array.isArray(data)) {
+          throw Error("Data is not an array");
+        }
+
+        setOperationList(data);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        setHasError(true);
+        setCurrentError(err.response?.data?.error);
+        setIsLoading(false);
+      });
+  }, [currentProject]);
+
+  if (isLoading) {
+    return (
+      <Placeholder>
+        <Loading />
+      </Placeholder>
+    );
+  }
+
+  if (operationList.length == 0) {
+    return <Placeholder>No operations available</Placeholder>;
+  }
+
+  if (hasError) {
+    return <Placeholder>Error</Placeholder>;
+  }
+
+  const getOperationDescription = (type: OperationType): string => {
+    switch (type) {
+      case "create":
+        return "This infrastructure was created";
+      case "update":
+        return "The infrastructure configuration was updated";
+      default:
+        return "The infrastructure configuration was updated";
```

Comment:
Super minor but we can avoid an unnecessary return here.

```suggestion
      case "update":
      default:
        return "The infrastructure configuration was updated";
```